File: Data Structures & Algorithms/max-consecutive-ones-iii/submission-0.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int i = 0;
        int j = 0;
        int zc = 0;
        int ans = 0;
        while(i < nums.size()){
            if (nums[i] == 0){
                zc++;
            }
            while (zc > k){
                if (nums[j] == 0){
                    zc--;
                } j++;


            }
            ans = max(ans,i-j+1);
            i++;
        }
        return ans;
        
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {

        // Step 1: frequency map
        Map<Integer, Integer> freq = new HashMap<>();
        for (int num : nums) {
            freq.put(num, freq.getOrDefault(num, 0) + 1);
        }

        // Step 2: Min Heap
        PriorityQueue<Integer> minHeap = new PriorityQueue<>(
            (a, b) -> freq.get(a) - freq.get(b)
        );

        // Step 3: offer & poll
        for (int num : freq.keySet()) {
            minHeap.offer(num);
            if (minHeap.size() > k) {
                minHeap.poll();
            }
        }

        // Step 4: result
        int[] result = new int[k];
        for (int i = k - 1; i >= 0; i--) {
            result[i] = minHeap.poll();
        }

        return result;
    }
}